core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -52,6 +53,7 @@ class BlockManagerMasterEndpoint(
     externalBlockStoreClient: Option[ExternalBlockStoreClient],
```

Comment:
Earlier, `externalBlockStoreClient` would be nonEmpty only when `SHUFFLE_SERVICE_FETCH_RDD_ENABLED` was enabled - but now it is nonEmpty in case shuffle service is enabled.
This results in `removeRdd` enabling `externalBlockStoreClient` related codepath.

For now, let us guard this with a check against `externalShuffleServiceRddFetchEnabled` and revisit enabling them in a different PR ?

Thoughts ?

Reply:
Ah yep good call, probably shouldn't be any blocks to delete but doesn't hurt to add that check back in